zipline/pipeline/loaders/blaze/core.py

```diff
@@ -311,22 +317,16 @@ def new_dataset(expr, deltas, missing_values):
         # Terms.
         if name in (SID_FIELD_NAME, TS_FIELD_NAME):
             continue
-        try:
-            # TODO: This should support datetime and bool columns.
-            if promote(type_, float64, promote_option=False) != float64:
-                raise NotPipelineCompatible()
-            if isinstance(type_, Option):
-                type_ = type_.ty
-        except NotPipelineCompatible:
-            col = NonPipelineField(name, type_)
-        except TypeError:
-            col = NonNumpyField(name, type_)
-        else:
+        if isinstance(type_, Option):
```

Comment:
We could probably move the unwrapping of `Option` into `datashape_type_to_numpy`.